src/main/payment/main/G1HRPS/Payment.java

```diff
@@ -1,164 +1,277 @@
 package main.G1HRPS;
 
 import java.util.UUID;
+import java.util.List;
 
 public class Payment {
 	/**
 	 * Unique generated ID
 	 */
 	private final UUID payment_id_;
-	private int guest_id_;
+	private String guest_id_;
+	private int room_num_; 
 	private float discounts_;
 	private float tax_;
+	private float room_charges_;
+	private float rs_charges_;
 	private float bill_total_;
 	private PaymentStatus status_;
 
 	/**
 	 * Payment Constructor.
 	 * 
-	 * @param id         UUID generated by manager
-	 * @param guest_id
-	 * @param discounts
-	 * @param bill_total After deducting discounts
-	 * @param status
+	 * @param id			UUID generated by manager
+	 * @param guest_id		Guest ID of guest that made payment
+	 * @param room_num		Room associated with this payment
+	 * @param discounts		Discount percentage in the range of (0.0-1.0)
+	 * @param tax			Tax on bill total in the range of (0.0-1.0)
+	 * @param bill_total	After deducting discounts
+	 * @param status		
 	 */
-	public Payment(UUID id, int guest_id, float discounts, float tax, float bill_total, PaymentStatus status) {
+	public Payment(UUID id, String guest_id, int room_num, float discounts, float tax, float bill_total, PaymentStatus status) {
 		payment_id_ = id;
 		guest_id_ = guest_id;
+		room_num_ = room_num;
 		discounts_ = discounts;
 		tax_ = tax;
+		room_charges_ = 0;
+		rs_charges_ = 0;
 		bill_total_ = bill_total;
 		status_ = status;
+	}
+
+	/**
+	 * 
+	 * @return	The payment_id of the payment
+	 */
+
+	public UUID GetPaymentID(){
+
+		return payment_id_;
 
 	}
 
 	/**
+	 * Assigns new guest_id to this payment
+	 * The guest with this guest_id has made this payment
 	 * 
 	 * @param guest_id
 	 */
-	public void SetGuestId(int guest_id) {
-		// TODO - implement Payment.SetGuestId
 
-		this.guest_id_ = guest_id;
+	public void SetGuestID(String guest_id) {
+
+		guest_id_ = guest_id;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public void SetDiscount(float discount) {
-		// TODO - implement Payment.SetDiscount
+	/**
+	 * 
```

Comment:
Add a brief here. e.g. Retrieves guest id of this guest. And these should end with a period.